pkg/testrig/testrig.go

```diff
@@ -0,0 +1,203 @@
+// Package testrig provides tools for generating standalone test environment for sysl generated services
+package testrig
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/anz-bank/sysl/pkg/syslutil"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/afero"
+)
+
+// GenerateRig creates full set of files required to start up a test rig
+// for every service, it creates main.go (optionally with DB connection) and a dockerfile to containerize it
+// if the service contains DB tables, it creates sidecar postgres container and creates a schema there
+// finally, it creates docker-compose.yml at the point of all (likely your project's root)
+// that joins all containerized services
+func GenerateRig(fs afero.Fs, templateFile string, outputDir string, modules []*sysl.Module) error {
+	serviceMap, err := readUserData(fs, templateFile)
+	if err != nil {
+		return err
+	}
+
+	applications := make(map[string]*sysl.Application)
+	for _, module := range modules {
+		for appName, app := range module.GetApps() {
+			applications[appName] = app
+		}
+	}
+
+	composeServices := make(map[string]interface{})
+
+	for serviceName, serviceTmpl := range serviceMap {
+		if err := fs.MkdirAll(filepath.Join(outputDir, serviceName), os.ModePerm); err != nil {
+			return err
+		}
+
+		if err := generateMain(fs, serviceName, serviceTmpl, outputDir, appNeedsDb(applications[serviceName])); err != nil {
+			return err
+		}
+
+		if err := generateDockerfile(fs, serviceName, serviceTmpl, outputDir); err != nil {
+			return err
+		}
+
+		if appNeedsDb(applications[serviceName]) {
+			composeServices[serviceName+"_db"] = generateDbService(serviceName)
+			composeServices[serviceName] = generateAppService(
```

Comment:
try to eliminate this repetition. perhaps:
```
var dependencies []string
if appNeedsDb(applications[serviceName]) {
        dbServiceName := serviceName + "_db"
        composeServices[dbServiceName] = generateDbService(serviceName)
        dependencies = []string{dbServiceName}
}
composeServices[serviceName] = generateAppService(&generateAppServiceInput{
        ServiceName: serviceName,
        Vars: serviceTmpl,
        OutputDir: outputDir,
        DependsOn: dependencies,
})
```

This keeps the common path to the left, and indents only the special case of the db service.